### PR TITLE
AG-8264 Add initial motion and animation manager

### PR DESCRIPTION
**Pull request by lsjroberts**
PR for https://ag-grid.atlassian.net/browse/AG-8264

Module example:

```ts
const rect = new _Scene.Rect();

ctx.animationManager.animate('example', {
    from: 0,
    to: Math.PI * 2,
    duration: 2000,
    ease: _Motion.linear,
    repeat: Infinity,
    repeatType: _Motion.RepeatType.Reverse,
    onUpdate: (v: number) => {
        rect.fill = `rgb(${255 - (v / (Math.PI * 2)) * 255}, 0, ${(v / (Math.PI * 2)) * 255})`;
        rect.x = 200 + v * 200;
        rect.y = 400 + Math.cos(v) * 200;
    },
});
```

### CODE REVIEW

**Review by alexanderby — APPROVED** [review]
👍

File: charts-community-modules/ag-charts-community/src/chart/interaction/animationManager.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                },
+                stop: () => {
+                    this.updaters.filter(([uid]) => uid !== id);
+                },
+            };
+        };
+    }
+
+    private requestAnimationFrame() {
+        this.requestId = requestAnimationFrame((time) => {
```
Comment (alexanderby) [inline charts-community-modules/ag-charts-community/src/chart/interaction/animationManager.ts:87]: One potential problem is, when the tab is hidden (`document.hidden`), nothing will happen. So `requestIdleCallback` should be there for a fallback.

**Comment by lsjroberts** [comment]
There are a few aspects to the integration with highlighting that are less than ideal and will need discussion before anything is merged.

**Comment by lsjroberts** [comment]
Closing this draft to recreate with cleaned up code and focused on just highlighting.

**Pull request closed without merging**